src/Command/Environment/AlignCommand.php

```diff
@@ -0,0 +1,190 @@
+
+        $search = $this->createSearch();
+        $scroll = $this->elasticaService->scroll($search, $this->scrollTimeout);
+        $total = $this->elasticaService->count($search);
+
+        $this->io->note(\sprintf('The source environment contains %s elements, start aligning environments...', $total));
+        $this->io->progressStart($total);
+
+        $deletedRevision = 0;
+        $alreadyAligned = 0;
+        $targetIsPreviewEnvironment = [];
+
+        foreach ($scroll as $resultSet) {
+            $this->publishService->bulkPublishStart();
+            $revisions = $this->revisionService->searchByResultSetEnvironment($resultSet, $this->source);
+
+            foreach ($revisions as $revision) {
+                $contentType = $revision->giveContentType();
+
+                if ($revision->getDeleted()) {
+                    ++$deletedRevision;
+                } elseif ($contentType->giveEnvironment()->getName() === $this->target->getName()) {
+                    if (!isset($targetIsPreviewEnvironment[$contentType->getName()])) {
+                        $targetIsPreviewEnvironment[$contentType->getName()] = 0;
+                    }
+                    ++$targetIsPreviewEnvironment[$contentType->getName()];
+                } else {
+                    if (0 == $this->publishService->bulkPublish($revision, $this->target)) {
+                        ++$alreadyAligned;
+                    }
+                }
+
+                $this->io->progressAdvance();
+            }
+            $this->publishService->bulkPublishFinished();
+        }
+
+        $this->io->progressFinish();
+
+        if ($input->getOption(self::OPTION_SNAPSHOT)) {
+            $this->environmentService->setSnapshotTag($this->target);
+            $this->io->note(\sprintf('The target environment "%s" was tagged as a snapshot', $this->target->getName()));
+        }
+
+        if ($deletedRevision > 0) {
+            $this->io->caution(\sprintf('%s deleted revisions were not aligned', $deletedRevision));
+        }
+
+        if ($alreadyAligned > 0) {
+            $this->io->note(\sprintf('%s revisions were already aligned', $alreadyAligned));
+        }
+
+        foreach ($targetIsPreviewEnvironment as $ctName => $counter) {
+            $this->io->caution(\sprintf(
+                '%s %s revisions were not aligned as %s is the default environment',
+                $counter, $ctName, $this->target->getName()
+            ));
+        }
+
+        $this->io->success(\vsprintf('Environments %s -> %s were aligned.', [
+            $this->source->getName(),
+            $this->target->getName(),
+        ]));
+
+        return self::EXECUTE_SUCCESS;
+    }
+
+    private function createSearch(): Search
+    {
+        $search = $this->elasticaService->convertElasticsearchBody(
+            [$this->source->getAlias()],
+            [],
+            ['query' => Json::decode($this->searchQuery)]
+        );
+        $search->setSize($this->scrollSize);
```

Comment:
Do you specify any _source fieldds? I think thta this ;etho should only get _id and _contenttype fields

Reply:
👍

Reply:
Good remark, indeed we only need _id and _contentType